db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -7339,6 +7342,8 @@ protected TabInfoImpl getNonCoreTIByNumber(int catalogNumber) throws StandardExc
             }
             initSystemIndexVariables(retval);
             noncoreInfo[nonCoreNum]=retval;
+            // Lazy Path for database re-boot
+            populateSystemConglomerates(new TabInfoImpl[]{retval});
```

Comment:
I don't think this is enough, at least for Vacuum. For vacuum we access the list of HBase conglomerates and then we'd use BasicDatabase.isSystemConglomerate() API, but some of these conglomerates might not have been lazily loaded.